src/Groups/libraries/libraries.jl

```diff
@@ -0,0 +1,87 @@
+export
+    moved_points,
+    number_moved_points
+
+moved_points(x::PermGroupElem) = [y for y in GAP.gap_to_julia(GAP.Globals.MovedPoints(x.X))]
+moved_points(G::PermGroup) = [y for y in GAP.gap_to_julia(GAP.Globals.MovedPoints(G.X))]
+number_moved_points(x::Union{PermGroupElem,PermGroup}) = length(moved_points(x))
+
+include("perfectgroups.jl")
+include("primitivegroups.jl")
+include("smallgroups.jl")
+include("transitivegroups.jl")
+
+#############################################################################
+
+# useful methods for all functions of type all_"property"_groups
+
+#############################################################################
+
+# return the output of the function f and the corresponding GAP function
+function find_index_function(f, isapg)
+   if f == isabelian return Bool, GAP.Globals.IsAbelian
+   elseif f == isalmostsimple return Bool, GAP.Globals.IsAlmostSimpleGroup
+   elseif f == iscyclic return Bool, GAP.Globals.IsCyclic
+   elseif f == isnilpotent return Bool, GAP.Globals.IsNilpotentGroup
+   elseif f == isperfect return Bool, GAP.Globals.IsPerfectGroup
+   elseif f == issimple return Bool, GAP.Globals.IsSimpleGroup
+   elseif f == issolvable return Bool, GAP.Globals.IsSolvableGroup
+   elseif f == issupersolvable return Bool, GAP.Globals.IsSupersolvableGroup
+   elseif f == istransitive return Bool, GAP.Globals.IsTransitive
+   elseif f == number_moved_points || f == degree
+      if isapg
+         return Union{Int, AbstractVector{Int}}, GAP.Globals.NrMovedPoints
```

Comment:
Why `AbstractVector{Int}`? In which situation does that happen?

Reply:
One can write something like `all_transitive_groups(degree, 4:6)` or `all_transitive_groups(degree,[4,5,6])` to get all transitive groups acting over a set of dimension 4, 5 or 6.
If the vector consists in just one point, then it is possible to write `all_transitive_groups(degree,5)` instead of `all_transitive_groups(degree,[5])`, similarly to how GAP works.

Reply:
I see. All this should ideally be documented, of course